scripts/scil_split_image.py

```diff
@@ -0,0 +1,147 @@
+
+    p.add_argument('--out_dwi', nargs='+', default=[],
+                   help='The names of the output DWI files. There must be '
+                        'one more name than indices in split_indices. By '
+                        'default, indices number will be appended to in_dwi, '
+                        'such as in_dwi_0_3, in_dwi_3_10, in_dwi_10_end.')
+
+    p.add_argument('--out_bval', nargs='+', default=[],
+                   help='The names of the output b-value files (.bval). There '
+                        'must be one more name than indices in split_indices. '
+                        'By default, indices number will be appended to '
+                        'in_dwi, such as in_dwi_0_3, in_dwi_3_10, '
+                        'in_dwi_10_end.')
+
+    p.add_argument('--out_bvec', nargs='+', default=[],
+                   help='The names of the output b-vector files (.bvec).There '
+                        'must be one more name than indices in split_indices. '
+                        'By default, indices number will be appended to '
+                        'in_dwi, such as in_dwi_0_3, in_dwi_3_10, '
+                        'in_dwi_10_end.')
+
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def create_name_from_indices(indices, input):
+    input_path = Path(input)
+    parent_dir = input_path.parent
+    input_name = Path(input_path.stem).stem
+    index_name = "_" + str(indices[0]) + "_" + str(indices[1])
+    suffix = ""
+    for suff in input_path.suffixes:
+        suffix += suff
+    output_name = Path(parent_dir, input_name + index_name + suffix)
+
+    return str(output_name)
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO)
+
+    assert_inputs_exist(parser, [args.in_dwi, args.in_bval, args.in_bvec])
+    assert_outputs_exist(parser, args, [],
+                         optional=list(np.concatenate((args.out_dwi,
+                                                       args.out_bval,
+                                                       args.out_bvec))))
+
+    # Check if the number of names given is equal to the number of indices + 1
+    if args.out_dwi and len(args.out_dwi) != len(args.split_indices) + 1:
+        parser.error('--out_dwi must contain len(split_indices) + 1 names.')
+    if args.out_bval and len(args.out_bval) != len(args.split_indices) + 1:
+        parser.error('--out_bval must contain len(split_indices) + 1 names.')
+    if args.out_bvec and len(args.out_bvec) != len(args.split_indices) + 1:
+        parser.error('--out_bvec must contain len(split_indices) + 1 names.')
+
+    bvals, bvecs = read_bvals_bvecs(args.in_bval, args.in_bvec)
+
+    img = nib.load(args.in_dwi)
+    data = img.get_fdata(dtype=np.float32)
+
+    # Check if the indices fit inside the range of possible values
+    if np.max(args.split_indices) > data.shape[-1]:
```

Comment:
Can you add a check if the indices are in increasing order?
Just in case someone tries to break it?